relentless/optimize/criteria.py

```diff
@@ -0,0 +1,427 @@
+        super().__init__(tolerance)
+
+    def converged(self, result):
+        """Check if the function is converged using the value test.
+
+        Parameters
+        ----------
+        result : :class:`~relentless.optimize.objective.ObjectiveFunctionResult`
+            The location of the function at which to check for convergence.
+
+        Returns
+        -------
+        bool
+            ``True`` if the function is converged.
+
+        """
+        converged = True
+        for x in result.design_variables:
+            tol = self._get_tol(x)
+            if x.value > tol:
+                converged = False
+                break
+
+        return converged
+
+class LogicTest(ConvergenceTest):
+    """Abstract base class for logical convergence tests.
+
+    Parameters
+    ----------
+    tests : args
+        The :class:`ConvergenceTest`\s to be used.
+
+    Raises
+    ------
+    TypeError
+        If all inputs are not :class:`ConvergenceTest`\s.
+
+    """
+    def __init__(self, *tests):
+        if not all([isinstance(t, ConvergenceTest) for t in tests]):
+            raise TypeError('All inputs to a LogicTest must be ConvergenceTests.')
+        self.tests = tests
+
+class AnyTest(LogicTest):
+    """Logic test if any specified test returns convergence.
+
+    Check if the function is determined to be converged by any of the specified
+    convergence tests.
+
+    Parameters
+    ----------
+    tests : args
+        The :class:`ConvergenceTest`\s to be used.
+
+    """
+    def __init__(self, *tests):
+        super().__init__(*tests)
+
+    def converged(self, result):
+        """Check if the function has converged by any of the specified tests.
+
+        Parameters
+        ----------
+        result : :class:`~relentless.optimize.objective.ObjectiveFunctionResult`
+            The location of the function at which to check for convergence.
+
+        Returns
+        -------
+        bool
+            ``True`` if the function is converged by any test.
+
+        """
+        return any([t.converged(result) for t in self.tests])
+
+class AllTest(LogicTest):
+    """Logic test if all specified tests return convergence.
+
+    Check if the function is determined to be converged by all of the specified
+    convergence tests.
+
+    Parameters
+    ----------
+    tests : args
+        The :class:`ConvergenceTest`\s to be used.
+
+    """
+    def __init__(self, *tests):
+        super().__init__(*tests)
+
+    def converged(self, result):
+        """Check if the function is converged by all of the specified tests.
+
+        Parameters
+        ----------
+        result : :class:`~relentless.optimize.objective.ObjectiveFunctionResult`
+            The location of the function at which to check for convergence.
+
+        Returns
+        -------
+        bool
+            ``True`` if the function is converged by all tests.
+
+        """
+        return all([t.converged(result) for t in self.tests])
```

Comment:
They are just special cases, but people might expect to see `AndTest` and `OrTest`. I think you can add them like this:
```py
class AndTest(AllTest):
    def __init__(self, a, b):
        super().__init__(a,b)
class OrTest(AnyTest):
    def __init__(self, a, b):
        super().__init__(a,b)
```